In [1]:
from google.colab import files
uploaded = files.upload()

Saving usuarios_win_mac_lin.csv to usuarios_win_mac_lin.csv


In [2]:
import pandas as pd

df = pd.read_csv("usuarios_win_mac_lin.csv")
print(df.head())

   duracion  paginas  acciones  valor  clase
0       7.0        2         4      8      2
1      21.0        2         6      6      2
2      57.0        2         4      4      2
3     101.0        3         6     12      2
4     109.0        2         6     12      2


In [3]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report, accuracy_score

# Variables
X = df.drop("clase", axis=1)
Y = df["clase"]

# Dividir datos
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, random_state=42)

# Modelo
modelo = LogisticRegression(max_iter=1000)

# Entrenar
modelo.fit(X_train, Y_train)

# Predicción
Y_pred = modelo.predict(X_test)

# Resultados
print("Precisión:", accuracy_score(Y_test, Y_pred))
print("\nReporte:\n", classification_report(Y_test, Y_pred))

Precisión: 0.6764705882352942

Reporte:
               precision    recall  f1-score   support

           0       0.62      0.81      0.70        16
           1       0.67      0.33      0.44        12
           2       0.86      1.00      0.92         6

    accuracy                           0.68        34
   macro avg       0.71      0.72      0.69        34
weighted avg       0.68      0.68      0.65        34



El modelo logró una precisión del 67%, lo que indica que puede clasificar correctamente la mayoría de los casos. Algunas clases se predicen mejor que otras, pero en general el modelo funciona de manera aceptable.